### DISCUSSION TITLE
How to build a model of two blocks without gaps？

### DISCUSSION THREAD

**Discussion by DYLDYLDYL** [idaholab/moose#22889] [Q&A Meshing] [answered]
Hi!
I wonder how to build a model of two blocks without gaps？My mesh is merged by  two mesh files with ICEM.
![Snipaste_2022-12-06_08-18-53](https://user-images.githubusercontent.com/91199452/205864388-49a29640-73af-4360-badc-b05f44cd34b6.jpg)

I want to simulate heat conduction between two blocks. And when I use GapHeatTransfer, it`s not work in this no-gap condition.
When adding Tensor Mechanics, it has gaps between blocks.
![2022-12-06 16-34-09屏幕截图](https://user-images.githubusercontent.com/91199452/205867292-7d7b7ef1-842b-4005-9e1e-793739c2a764.png)

I wonder how to creat a whole  Heat Conduction and Tensor Mechanics  Mesh with tow different materials blocks.

Thanks in advance for your help!

**Comment by Edward-Eth** [comment, marked as answer]
If the nodes on each mesh align you could use the StitchedMeshGenerator to merge the nodes of the two meshes, making them connected for the physics 👍

**Reply by DYLDYLDYL** [reply]
Thanks a lot ！Your answer solves my problem perfectly ！

**Comment by GiudGiud** [comment]
Hello

You may want to use ThinLayerHeatTransfer in the connected situation instead of the GapHeatTransfer
https://mooseframework.inl.gov/moose/source/interfacekernels/ThinLayerHeatTransfer.html

Guillaume

**Reply by DYLDYLDYL** [reply]
Thanks a lot ！ I use the StitchedMeshGenerator, set heat source in block 1, and find heat transfer in block 2.  And when adding Tensor Mechanics, it has no gaps between blocks! My problem is solved perfectly!
